src/PKSim.Core/Services/PKAnalysesTask.cs

```diff
@@ -177,24 +178,78 @@ public GlobalPKAnalysis CalculateGlobalPKAnalysisFor(IEnumerable<Simulation> sim
          if (simulation == null)
             return globalPKAnalysis;
 
-         simulation.Compounds.Each(compound =>
+         // For a population simulation the pk parameters are calculated at run time so we retrieve the global
+         // pk parameters from the simulation
+         if (simulation is PopulationSimulation populationSimulation)
          {
-            var pkCalculationOptions = _pkCalculationOptionsFactory.CreateFor(simulation, compound.Name);
-            var peripheralVenousBloodColumn = simulation.PeripheralVenousBloodColumn(compound.Name);
-            var venousBloodColumn = simulation.VenousBloodColumn(compound.Name);
-            var compoundContainer = calculateGlobalPKAnalysisFor(simulation, compound, peripheralVenousBloodColumn, venousBloodColumn, pkCalculationOptions);
-            globalPKAnalysis.Add(compoundContainer);
-         });
+            // Global PK Parameters are stored in containers with the compound name as the first child of the root container.
+            // There will be one container per compound with PK parameters stored within
+            var compoundContainers = createCompoundPKParameterContainers(populationSimulation);
+            compoundContainers.Each(globalPKAnalysis.Add);
+
+            return globalPKAnalysis;
+         }
+
+         // For individual simulations global pk parameters are calculated on the fly
+         if (simulation is IndividualSimulation individualSimulation)
+         {
+            individualSimulation.Compounds.Each(compound =>
+            {
+               var pkCalculationOptions = _pkCalculationOptionsFactory.CreateFor(individualSimulation, compound.Name);
+               var peripheralVenousBloodColumn = individualSimulation.PeripheralVenousBloodColumn(compound.Name);
+               var venousBloodColumn = individualSimulation.VenousBloodColumn(compound.Name);
+               var individualId = individualSimulation.Results.AllIndividualIds().FirstOrDefault();
```

Comment:
I don't like this. Can we create a method in the indiviudalSimulation? This is an ecampsulation details that is leaking here.

Reply:
Do you mean ask the IndividualSimulation for it's IndividualId rather than having everyone know that the IndividualSimulation has an Enumerable of which there is only 1?

Reply:
yes. Suddently we know that the simulation as this REsults that look like a population etc..